tests/test_parameters.py

```diff
@@ -0,0 +1,181 @@
+import numpy as np
+import pytest
+import pyfar as pf
+import pyrato as ra
+from pyrato.parameters import clarity
+import numpy.testing as npt
+import re
+
+
+def test_fixture_returns_timedata_object(make_edc_from_energy):
+    edc = make_edc_from_energy()
+    assert isinstance(edc, pf.TimeData)
+    assert edc.time.ndim >= 1  # has time dimension
+    assert np.all(edc.time.shape)  # not empty
+
+def test_fixture_preserves_input_shape(make_edc_from_energy):
+    custom_energy = np.ones((2, 3, 10))
+    edc = make_edc_from_energy(energy=custom_energy)
+    assert edc.time.shape == custom_energy.shape
+
+@pytest.mark.parametrize("invalid_energy", ["not_an_array", 5, None])
+def test_fixture_rejects_invalid_energy_type(make_edc_from_energy,
+                                             invalid_energy):
+    if invalid_energy is None:
+        pytest.skip("None is allowed and defaults to baseline")
+    with pytest.raises(TypeError):
+        make_edc_from_energy(energy=invalid_energy)
+
+def test_fixture_handles_zero_energy(make_edc_from_energy):
+    snr = 90  # -90 dBFS
+    edc = make_edc_from_energy(energy=np.zeros(10), dynamic_range=snr)
+    assert np.allclose(edc.time, 10 ** (-snr / 10))
```

Comment:
```suggestion
    dynamic_range = 90  # -90 dBFS
    edc = make_edc_from_energy(
        energy=np.zeros(10), dynamic_range=dynamic_range)
    assert np.allclose(edc.time, 10 ** (-dynamic_range / 10))
```
The term SNR has another meaning and could be confusing here